File: android-app/baselibrary/src/main/java/com/musketeer/baselibrary/util/SharePreferenceUtils.java
/**   
 * @Title: SharePreferenceUtils.java 
 * @Package com.musketeer.lib.util 
 *
 * @author musketeer zhongxuqi@163.com  
 * @date 2014-11-12 下午1:51:30 
 * @version V1.0   
 */
package com.musketeer.baselibrary.util;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * @author zhongxuqi
 * 
 */
public class SharePreferenceUtils {
	public static final String PREFERENCE_NAME = "Musketeer";

	public static boolean getBoolean(Context context, String key,
			boolean defaultvalue) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		return mSharedPreferences.getBoolean(key, defaultvalue);
	}

	public static boolean putBoolean(Context context, String key, boolean value) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		SharedPreferences.Editor editor = mSharedPreferences.edit();
		editor.putBoolean(key, value);
		return editor.commit();
	}

	public static String getString(Context context, String key,
			String defaultvalue) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		return mSharedPreferences.getString(key, defaultvalue);
	}

	public static boolean putString(Context context, String key, String value) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		SharedPreferences.Editor editor = mSharedPreferences.edit();
		editor.putString(key, value);
		return editor.commit();
	}

	public static int getInt(Context context, String key, int defaultvalue) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		return mSharedPreferences.getInt(key, defaultvalue);
	}

	public static boolean putInt(Context context, String key, int value) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		SharedPreferences.Editor editor = mSharedPreferences.edit();
		editor.putInt(key, value);
		return editor.commit();
	}

	public static long getLong(Context context, String key, long defaultvalue) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		return mSharedPreferences.getLong(key, defaultvalue);
	}

	public static boolean putLong(Context context, String key, long value) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		SharedPreferences.Editor editor = mSharedPreferences.edit();
		editor.putLong(key, value);
		return editor.commit();
	}

	public static float getFloat(Context context, String key, float defaultvalue) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		return mSharedPreferences.getFloat(key, defaultvalue);
	}

	public static boolean putFloat(Context context, String key, float value) {
		SharedPreferences mSharedPreferences = context.getSharedPreferences(
				PREFERENCE_NAME, Context.MODE_PRIVATE);
		SharedPreferences.Editor editor = mSharedPreferences.edit();
		editor.putFloat(key, value);
		return editor.commit();
	}

}

File: android-app/app/src/main/kotlin/com/musketeer/scratchpaper/utils/SharePreferenceUtils.kt
/**
 * @Title: SharePreferenceUtils.java
 * *
 * @Package com.musketeer.lib.util
 * *
 * *
 * @author musketeer zhongxuqi@163.com
 * *
 * @date 2014-11-12 下午1:51:30
 * *
 * @version V1.0
 */
package com.musketeer.scratchpaper.utils

import android.content.Context
import android.content.SharedPreferences

/**
 * @author zhongxuqi
 */
object SharePreferenceUtils {
    val PREFERENCE_NAME = "Musketeer"

    fun getBoolean(context: Context, key: String,
                   defaultvalue: Boolean): Boolean {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        return mSharedPreferences.getBoolean(key, defaultvalue)
    }

    fun putBoolean(context: Context, key: String, value: Boolean): Boolean {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        val editor = mSharedPreferences.edit()
        editor.putBoolean(key, value)
        return editor.commit()
    }

    fun getString(context: Context, key: String,
                  defaultvalue: String): String {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        return mSharedPreferences.getString(key, defaultvalue)
    }

    fun putString(context: Context, key: String, value: String): Boolean {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        val editor = mSharedPreferences.edit()
        editor.putString(key, value)
        return editor.commit()
    }

    fun getInt(context: Context, key: String, defaultvalue: Int): Int {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        return mSharedPreferences.getInt(key, defaultvalue)
    }

    fun putInt(context: Context, key: String, value: Int): Boolean {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        val editor = mSharedPreferences.edit()
        editor.putInt(key, value)
        return editor.commit()
    }

    fun getLong(context: Context, key: String, defaultvalue: Long): Long {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        return mSharedPreferences.getLong(key, defaultvalue)
    }

    fun putLong(context: Context, key: String, value: Long): Boolean {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        val editor = mSharedPreferences.edit()
        editor.putLong(key, value)
        return editor.commit()
    }

    fun getFloat(context: Context, key: String, defaultvalue: Float): Float {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        return mSharedPreferences.getFloat(key, defaultvalue)
    }

    fun putFloat(context: Context, key: String, value: Float): Boolean {
        val mSharedPreferences = context.getSharedPreferences(
                PREFERENCE_NAME, Context.MODE_PRIVATE)
        val editor = mSharedPreferences.edit()
        editor.putFloat(key, value)
        return editor.commit()
    }

}
